scripts/build-all-in-one-image.sh

```diff
@@ -45,17 +45,108 @@ upload_to_docker() {
   fi
 }
 
-make build-all-in-one GOOS=linux GOARCH=$GOARCH
+upload_multiarch_to_docker() {
+  # Only push the docker image to dockerhub/quay.io for master/release branch
+  if [[ "$BRANCH" == "master" || $BRANCH =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo "upload $1 to dockerhub/quay.io"
+    REPO=$1
+    build_upload_multiarch_to_docker $REPO
+  else
+    echo 'skip docker images upload for PR'
+  fi
+}
+
+build_upload_multiarch_to_docker(){
+  DOCKERHUB_USERNAME=${DOCKERHUB_USERNAME:-"jaegertracingbot"}
+  DOCKERHUB_TOKEN=${DOCKERHUB_TOKEN:-}
+  QUAY_USERNAME=${QUAY_USERNAME:-"jaegertracing+github_workflows"}
+  QUAY_TOKEN=${QUAY_TOKEN:-}
+
+  ###############Compute the tag
+  BASE_BUILD_IMAGE=$1
+
+  ## if we are on a release tag, let's extract the version number
+  ## the other possible value, currently, is 'master' (or another branch name)
+  if [[ $BRANCH == v* ]]; then
+      COMPONENT_VERSION=$(echo ${BRANCH} | grep -Po "([\d\.]+)")
+      MAJOR_MINOR=$(echo ${COMPONENT_VERSION} | awk -F. '{print $1"."$2}')
```

Comment:
The `COMPONENT_VERSION` variable in `L71` is actually for patch components, like `1.22.0`. I've changed the name of this variable to `MAJOR_MINOR_PATCH` to avoid ambiguity.